rosbag2_storage/include/rosbag2_storage/yaml.hpp

```
@@ -42,7 +42,18 @@ template<typename T>
 void optional_assign(const Node & node, std::string field, T & assign_to)
 {
   if (node[field]) {
-    YAML::convert<T>::decode(node[field], assign_to);
+    try {
+      if (!YAML::convert<T>::decode(node[field], assign_to)) {
+        throw YAML::Exception(
+          node[field].Mark(),
+          "Failed to convert field '" + field + "'");
+      }
+    } catch (const YAML::Exception & ex) {
+      // Preserve original error but add field name context
+      throw YAML::Exception(
+        node[field].Mark(),
+        "Failed to convert field '" + field + "': " + ex.what());
+    }
```

@fujitatomoya I am a bit confused about why you need to throw an exception, then catch it, and then rethrow it again? Can you please explain?

Maybe, would be clearer to throw an inner runtime exception, catch it and wrap it in the `YAML::Exception(..)`?